extensions/wikia/AdMixExperiment/scripts/AdMixExperiment.js

```diff
@@ -0,0 +1,294 @@
+					position: '',
+					top: '',
+					width: ''
+				});
+			}
+
+			resetAdAndRecircWrapper();
+		}
+
+		function resetTopAd() {
+			$topRightAd.css({
+				position: '',
+				top: '',
+				width: ''
+			});
+		}
+
+		function resetTopAdPaddings() {
+			$topRightAdWrapper.css({
+				'padding-top': '',
+				'padding-bottom': gapSize + 'px'
+			});
+		}
+
+		function resetAdAndRecircWrapper() {
+			$adAndRecircWrapper.css({
+				bottom: '',
+				position: '',
+				top: '',
+				width: ''
+			});
+		}
+
+		function update() {
+			var scrollTop = $window.scrollTop();
+			var firstAdTopPosition = getFirstAdTopPosition();
+			var firstAdBottomPosition = getFirstAdBottomPosition(firstAdTopPosition);
+			var adAndRecircWrapperTopPosition = getAdAndRecircWrapperTopPosition();
+			var adAndRecircWrapperBottomPosition = getAdAndRecircWrapperBottomPosition(adAndRecircWrapperTopPosition);
+			var stopPosition = getStopPosition();
+
+			if (
+				scrollTop < firstAdTopPosition ||
+				(!topRightAdFixed && scrollTop < adAndRecircWrapperTopPosition) ||
+				viewportWidth <= breakpointSmall
+			) {
+				reset();
+			} else if (
+				scrollTop >= firstAdTopPosition &&
+				scrollTop < firstAdBottomPosition
+			) {
+				scrollAfterFirstAdTopPosition();
+			} else if (
+				scrollTop >= firstAdBottomPosition &&
+				scrollTop < adAndRecircWrapperTopPosition
+			) {
+				scrollAfterFirstAdBottomPosition();
+			} else if (
+				scrollTop >= adAndRecircWrapperTopPosition &&
+				scrollTop < adAndRecircWrapperBottomPosition
+			) {
+				scrollAfterAdAndRecircWrapperTopPosition();
+			} else if (
+				scrollTop >= adAndRecircWrapperBottomPosition &&
+				scrollTop < stopPosition
+			) {
+				scrollAfterAdAndRecircWrapperBottomPosition();
+			} else if (scrollTop >= stopPosition) {
+				scrollStopPosition(stopPosition);
+			}
+		}
+
+		function onRightRailReady() {
+			$recircWrapper = $('#recirculation-rail');
+
+			// There is no recirculation right rail module on non-en wikis
+			if (window.wgContentLanguage === 'en') {
+				$recircWrapper.one('premiumRecirculationRail.ready', onRecircReady);
+			} else {
+				recircEnabled = false;
+				gapSize = 0;
+				onRecircReady();
+			}
+		}
+
+		function onRecircReady() {
+			if (recircEnabled) {
+				$recirc = $recircWrapper.find('.premium-recirculation-rail');
+				$recircWrapper
+					.css('margin-bottom', gapSize + 'px')
+					.height($recirc.outerHeight(true));
+			}
+
+			$adAndRecircWrapper = $('#WikiaAdInContentPlaceHolder');
+
+			$visibleElementBeforeWrapper = $adAndRecircWrapper.prevAll(':not(#LEFT_SKYSCRAPER_2)').eq(0);
+			visibleElementBeforeWrapperHeight = $visibleElementBeforeWrapper.outerHeight(true);
+
+			$window.scroll(throttle(update, 100));
+			$window.resize(throttle(function () {
+				viewportHeight = $(window).height();
+				viewportWidth = $(window).width();
+				update();
+			}, 100));
+
+			update();
+		}
+
+		$rail.one('afterLoad.rail', onRightRailReady);
+
+		if (topRightAdFixed) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Change this condition so that it does not always evaluate to "false"; some subsequent code is never executed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2583)